cli/ops/io.rs

```diff
@@ -150,10 +213,22 @@ pub fn op_read(
 
   poll_fn(move |cx| {
     let resource_table = &mut state.borrow_mut().resource_table;
-    let resource = resource_table
-      .get_mut::<StreamResource>(rid as u32)
+    let resource_holder = resource_table
+      .get_mut::<StreamResourceHolder>(rid as u32)
       .ok_or_else(OpError::bad_resource_id)?;
-    let nread = ready!(resource.poll_read(cx, &mut buf.as_mut()[..]))?;
+    let nread = match resource_holder
+      .resource
+      .poll_read(cx, &mut buf.as_mut()[..])
+    {
+      Poll::Ready(t) => {
+        resource_holder.untrack_task();
+        t
+      }
+      Poll::Pending => {
+        resource_holder.track_task(cx)?;
+        return Poll::Pending;
+      }
+    }?;
```

Comment:
@ry, @piscisaureus and I had a discussion about this solution; it looks good on the surface; but it effectively prevents having multiple ops reading from single resource. Although in most cases we only one to have a single op there are some cases where having multiple ops reading from single resource is desirable. 

Ideally we would track multiple tasks for single resource, but `Waker` is very simple struct that has no `PartialEq` or any other trait implementation that would allow to discriminate wakers.

Reply:
We'll discuss it more tonight

Reply:
@bartlomieju I pushed some small changes to the `StreamResourceWrapper` to make it capable of waking multiple tasks. It just a suggestion to help the discussion.